src/PowerShellEditorServices/Language/LanguageService.cs

```diff
@@ -569,6 +591,55 @@ public FindOccurrencesResult FindSymbolsInFile(ScriptFile scriptFile)
             return foundDefinition;
         }
 
+        private Ast FindSmallestStatementAst(ScriptFile scriptFile, int lineNumber, int columnNumber)
+        {
+            var asts = scriptFile.ScriptAst.FindAll(ast =>
+            {
+                if (!(ast is StatementAst))
+                {
+                    return false;
+                }
+
+                var scriptExtent = ast.Extent;
+                if (scriptExtent.StartLineNumber > lineNumber || scriptExtent.EndLineNumber < lineNumber)
+                {
+                    return false;
+                }
+
+                if (scriptExtent.StartLineNumber == lineNumber)
+                {
+                    return scriptExtent.StartColumnNumber <= columnNumber;
+                }
+
+                if (scriptExtent.EndLineNumber == lineNumber)
+                {
+                    return scriptExtent.EndColumnNumber >= columnNumber;
+                }
+
+                return true;
+            }, true);
+
+            if (asts == null || asts.Count() == 0)
+            {
+                return null;
+            }
+
+            Func<IScriptExtent, int> getExtentWitdh = extent => extent.EndOffset - extent.StartOffset;
+            var minDiff = getExtentWitdh(scriptFile.ScriptAst.Extent);
+            Ast minAst = scriptFile.ScriptAst;
+            foreach (var ast in asts)
```

Comment:
Can you add a comment explaining what this loop is doing?  Isn't perfectly clear why it's necessary.